service/services/services.go

```diff
@@ -0,0 +1,664 @@
+	if query == nil {
+		return []string{}, "", false, errors.New("Query is not present")
+	}
+	colStr := ""
+	if query.OnlyCount {
+		return []string{"count"}, "COUNT(" + pKey + ") AS count", true, nil
+	}
+	table := changeTableNameForSP(query.TableName)
+	var cols []string
+	if query.ProjectionExpression != "" {
+		cols = getSpannerProjections(query.ProjectionExpression, query.TableName, query.ExpressionAttributeNames)
+		insertpKey := true
+		for i := 0; i < len(cols); i++ {
+			if cols[i] == pKey {
+				insertpKey = false
+				break
+			}
+		}
+		if insertpKey {
+			cols = append(cols, pKey)
+		}
+		if sKey != "" {
+			insertSKey := true
+			for i := 0; i < len(cols); i++ {
+				if cols[i] == sKey {
+					insertSKey = false
+					break
+				}
+			}
+			if insertSKey {
+				cols = append(cols, sKey)
+			}
+		}
+		if tPkey != pKey {
+			insertSKey := true
+			for i := 0; i < len(cols); i++ {
+				if cols[i] == tPkey {
+					insertSKey = false
+					break
+				}
+			}
+			if insertSKey {
+				cols = append(cols, tPkey)
+			}
+		}
+
+	} else {
+		cols = models.TableColumnMap[table]
+	}
+	for i := 0; i < len(cols); i++ {
+		if cols[i] == "commit_timestamp" {
+			continue
+		}
+		colStr += table + ".`" + cols[i] + "`,"
+	}
+	colStr = strings.Trim(colStr, ",")
+	return cols, colStr, false, nil
+}
+
+func changeTableNameForSP(tableName string) string {
+	tableName = strings.ReplaceAll(tableName, "-", "_")
+	return tableName
+}
+
+func parseSpannerTableName(query *models.Query) string {
+	tableName := changeTableNameForSP(query.TableName)
+	if query.IndexName != "" {
+		tableName += "@{FORCE_INDEX=" + query.IndexName + "}"
+	}
+	return tableName
+}
+
+func parseSpannerCondition(query *models.Query, pKey, sKey string) (string, map[string]interface{}) {
+	params := make(map[string]interface{})
+	whereClause := "WHERE "
+
+	if sKey != "" {
+		whereClause += sKey + " is not null "
+	}
+
+	if query.RangeExp != "" {
+		if whereClause != "WHERE " {
+			whereClause += " AND "
+		}
+		_, _, query.RangeExp = utils.ParseBeginsWith(query.RangeExp)
+		query.RangeExp = strings.ReplaceAll(query.RangeExp, "begins_with", "STARTS_WITH")
+
+		count := 1
+		for k, v := range query.RangeValMap {
+			if strings.Contains(query.RangeExp, k) {
+				str := "rangeExp" + strconv.Itoa(count)
+				query.RangeExp = strings.ReplaceAll(query.RangeExp, k, "@"+str)
+				params[str] = v
+				count++
+			}
+		}
+		whereClause += query.RangeExp
+	}
+
+	if query.FilterExp != "" {
+		if whereClause != "WHERE " {
+			whereClause += " AND "
+		}
+		_, _, query.FilterExp = utils.ParseBeginsWith(query.FilterExp)
+		query.FilterExp = strings.ReplaceAll(query.FilterExp, "begins_with", "STARTS_WITH")
+		count := 1
+		for k, v := range query.RangeValMap {
+			if strings.Contains(query.FilterExp, k) {
+				str := "filter" + strconv.Itoa(count)
+				query.FilterExp = strings.ReplaceAll(query.FilterExp, k, "@"+str)
+				params[str] = v
+				count++
+			}
+		}
+		whereClause += query.FilterExp
+	}
```

Comment:
The code of above two`if` statements looks very similar, is it possible to extract the common part and make it a function?